app/src/main/java/com/imashnake/animite/data/Resource.kt

```diff
@@ -1,64 +1,42 @@
 package com.imashnake.animite.data
 
-import com.apollographql.apollo3.ApolloCall
-import com.apollographql.apollo3.api.ApolloResponse
-import com.apollographql.apollo3.api.Operation
-import com.apollographql.apollo3.exception.ApolloCompositeException
-import com.apollographql.apollo3.exception.ApolloNetworkException
-import com.apollographql.apollo3.exception.CacheMissException
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.catch
-import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapLatest
 
 sealed class Resource<T>(
     open val data: T?,
     open val message: String? = null
 ) {
 
     data class Success<T>(override val data: T) : Resource<T>(data)
-    data class Error<T>(override val message: String?, override val data: T? = null) : Resource<T?>(data, message)
+    data class Error<T>(override val message: String?, override val data: T? = null) : Resource<T>(data, message)
```

Comment:
oh that's how it's done, I see

Reply:
I didn't change the loading one, did you mean `Resource.Error<*>`?

Even then, the behavior as written shouldn't be any different from before. If anything, it's now less casting because we aren't managing a `Resource<T>` and a `Resource<T?>`